draco/analysis/delay.py

```diff
@@ -1082,18 +1082,27 @@ def _evaluate(self, data_view, weight_view, out_cont, delays, channel_ind):
             data = data_view.local_array[lbi]
             weight = weight_view.local_array[lbi]
 
+            # Apply the window if requested
+            # Do this before applying data cut
+            # as that will change the data shape
+            if self.apply_window:
+                data *= window
+
             # Apply data cuts
             t = self._cut_data(data, weight)
             if t is None:
                 continue
             data, _, _, nzt = t
 
-            # Apply the window if requested
-            if self.apply_window:
-                data *= window
+            # Take the inverse Fourier transform
+            # Note fftw only supports complex to
+            # complex transform. For real value ringmap
+            # use numpy fft.
+            if data.dtype is complex:
```

Comment:
```suggestion
            if np.iscomplexobj(data):
```

Numpy dtypes don't agree with python, so this will end up always returning False. Just use the `numpy` built-in instead.